ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/keyboard.html

```diff
@@ -180,6 +180,25 @@
                 }
             }
         
+            function suggestionPopup(obj,custom,x,y,w,h) {
+              if(obj != null) {
+                var i;
+                var s = JSON.stringify(obj);
+                var shift = false;
+                var pos = x.toString() + ',' + y.toString();
+                var size = w.toString() + ',' + h.toString();
+                var frame = pos + ',' + size;
+
+                fragmentToggle=(fragmentToggle+1) % 100;
+                var hash = 'suggestPopup-' + fragmentToggle + '+baseFrame=' + frame + '+suggestion=' + s;
```

Comment:
To benefit from using builtin URL parsers, change the `+`s to `&`s.

Reply:
I would also suggest using `encodeURIComponent` on each parameter value.

Reply:
Again, iOS isn't using a URL.

Reply:
Gotcha. Could things go wrong if there is an embedded `+` in the `s` object?